src/sentry/search/events/builder/metrics.py

```diff
@@ -772,6 +811,47 @@ def resolve_limit(self, limit: Optional[int]) -> Optional[Limit]:
     def resolve_granularity(self) -> Granularity:
         return Granularity(self._granularity)
 
+    def get_snql_query(self) -> Request:
+        # We are overriding this method here because in the case of alerts, we would like to use the snql query
+        # generated by the metrics layer. This implementation goes against the rationale of the metrics layer and should
+        # be removed as soon as snuba subscriptions will be supported by the layer.
+        # The logic behind this method is that MetricsQueryBuilder will generate a dialect of snql via the
+        # "get_metrics_layer_snql_query()" method, which will be fed into the transformer that will generate
+        # the MetricsQuery which is the DSL of querying that the metrics layer understands. Once this is generated,
+        # we are going to import the purposfully hidden SnubaQueryBuilder which is a component that takes a MetricsQuery
```

Comment:
nit, can this be a docstring instead?
```suggestion
        """
        We are overriding this method here because in the case of alerts, we would like to use the snql query
        generated by the metrics layer. This implementation goes against the rationale of the metrics layer and should
        be removed as soon as snuba subscriptions will be supported by the layer.
        The logic behind this method is that MetricsQueryBuilder will generate a dialect of snql via the
        "get_metrics_layer_snql_query()" method, which will be fed into the transformer that will generate
        the MetricsQuery which is the DSL of querying that the metrics layer understands. Once this is generated,
        we are going to import the purposfully hidden SnubaQueryBuilder which is a component that takes a MetricsQuery
        """
```